plugin/trino-hive/src/main/java/io/trino/plugin/hive/LegacyHiveViewReader.java

```diff
@@ -49,7 +49,7 @@ public ConnectorViewDefinition decodeViewData(String viewData, Table table, Cata
                         .map(column -> new ConnectorViewDefinition.ViewColumn(column.getName(), TypeId.of(column.getType().getTypeSignature().toString())))
                         .collect(toImmutableList()),
                 Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
-                table.getOwner(),
+                hiveViewsRunAsInvoker ? Optional.empty() : table.getOwner(),
```

Comment:
I think similar should be done in the non-legacy reader https://github.com/trinodb/trino/pull/12221/files#r970434417

Reply:
The PR changes `LegacyHiveViewReader` and this looks OK.
Doesn't it need to change `io.trino.plugin.hive.ViewReaderUtil.HiveViewReader#decodeViewData` too?

Reply:
The non-legacy reader does already handle setting the owner correctly. `HiveViewReader#decodeViewData` initially returns with an empty owner, but sets an owner, if using DEFINER and if the owner exists, in [HiveMetadata#getView](https://github.com/joechu1/trino/blob/f5ac41a8be77b22d2e8027a89decff44892f6099/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L2445). 
However, I think we can also just streamline this in `HiveViewReader#decodeViewData` and not need the additional logic  `HiveMetadata#getView`.

Reply:
Looks like we smeared the responsibility across 3 places:

- legacy HiveViewReader
- non-legacy HiveViewReader
- HiveMetadata#getView

and the result is quite inconsistent (hence questions like above)

Looking into this more, i no longer understand why the change here is desireable.
We have the `table.getOwner()` and it's some information, why do we need the LegacyHiveViewReader not to pass it further?

Reply:
I just updated the description in #12403 , but the problem comes from creating a [ConnectorViewDefinition](https://github.com/trinodb/trino/blob/dff92cf8a3ef8e1523a50c05e3cb4674acf3f465/core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorViewDefinition.java), which does not allow passing in an owner if running as invoker.

Reply:
OK, I get it.
But why would we pass `hiveViewsRunAsInvoker ? Optional.empty() : table.getOwner()` here and
`Optional.empty()` in non-legacy HiveViewReader?

HiveViewReader and LegacyHiveViewReader should differ in how they translate view SQL (with coral or not), but they should not differ in how they translate authorization/ownership, or whether the produce INVOKER or DEFINER view.

Reply:
That's part of the smeared responsibility that you mentioned. The non-legacy HiveViewReader passes an `Optional.empty()`, but afterwards `HiveMetadata#getView` will create another `ConnectorViewDefinition` object with the proper owner set, if needed. I feel the same, which is why I created the added the [extra commit](https://github.com/trinodb/trino/pull/12403/commits/834d03aac5dda1dc193fa215fac420924551f7a9). 
If there is agreement with this, perhaps we can add that commit to this PR as well.

Reply:
For now, let's have

```suggestion
                Optional.empty(), // will be filled in later by HiveMetadata
```

and same in the other class.

Would that work?

Reply:
No objections from me. I can confirm that the suggested change will also resolve the issue. What do you think @skrzypo987?

Reply:
That would definitely be more consistent. Pushed the new version already